src/core/execute.c

```diff
@@ -3582,11 +3582,14 @@ static int exec_child(
                 strv_free_and_replace(accum_env, ee);
         }
 
-        final_argv = replace_env_argv(command->argv, accum_env);
-        if (!final_argv) {
-                *exit_status = EXIT_MEMORY;
-                return log_oom();
-        }
+        if (!(command->flags & EXEC_COMMAND_NO_ENV)) {
+                final_argv = replace_env_argv(command->argv, accum_env);
+                if (!final_argv) {
+                        *exit_status = EXIT_MEMORY;
+                        return log_oom();
+                }
+        } else
+                final_argv = command->argv;
```

Comment:
final_argv is declared with _cleanup_strv_ which means there's a memory corruption here, as you will free command->argv if execution fails, but you really shouldn't, as that memor s owned by whoever owns `command`...

Please maybe drop the `_cleanup_strv_` from `final_argv` and introduce a new variable `replaced` or so which carries it. Then assign the result of `replace_env_argv` to both final_argv and to `replaced`, so that it is freed through the latter, but we still can reference it via `final_argv`...